test/e2e/admin/admin-checkout.test.ts

```diff
@@ -0,0 +1,317 @@
+    this.verifyCheckoutItem(res.body.body.data);
+  }
+
+  @test('successfully gets list of checked out items')
+  @slow(1500)
+  public async getCheckedOutItemsSuccessfully() {
+    // GIVEN: API
+    // WHEN: Getting the list of checked out items
+    const user = await loginAdmin();
+    const idToken = await user.getIdToken();
+    const res = await this.chai
+      .request(this.app)
+      .get(this.apiEndpoint)
+      .set('idToken', idToken)
+      .set('content-type', 'application/json');
+    // THEN: Returns a well formed response
+    super.assertRequestFormat(res);
+    // THEN: Locations are checked
+    await this.verifyCheckoutObjects(res.body.body.data);
+  }
+
+  @test('successfully gets list of available items')
+  @slow(1500)
+  public async getCheckoutItemsSuccessfully() {
+    // GIVEN: API
+    // WHEN: Getting the list of available items
+    const user = await loginAdmin();
+    const idToken = await user.getIdToken();
+    const res = await this.chai
+      .request(this.app)
+      .get(`${this.apiEndpoint}/items`)
+      .set('idToken', idToken)
+      .set('content-type', 'application/json');
+    // THEN: Returns a well formed response
+    super.assertRequestFormat(res);
+    // THEN: Locations are checked
+    await this.verifyCheckoutItems(res.body.body.data);
+  }
+
+  @test('fails to create new checkout due to no item id')
+  @slow(1500)
+  public async createNewCheckoutFailsDueToNoItemId() {
+    // GIVEN: API
+    // WHEN: Creating a new checkout object
+    const user = await loginAdmin();
+    const idToken = await user.getIdToken();
+    const res = await this.chai
+      .request(this.app)
+      .post(this.apiEndpoint)
+      .set('idToken', idToken)
+      .set('content-type', 'application/json');
+
+    // THEN: Returns a well formed response
+    super.assertRequestFormat(res, 'Error', 400, 'Error');
+    // THEN: Failed to validate input
+    this.expect(res.body.body.data).to.deep.equal({ message: 'Cannot find item ID to checkout' });
+  }
+
+  @test('fails to create new checkout due to no user id')
+  @slow(1500)
+  public async createNewCheckoutFailsDueToNoUserId() {
+    // GIVEN: API
+    // WHEN: Creating a new checkout object
+    const user = await loginAdmin();
+    const idToken = await user.getIdToken();
+    const parameters = {
+      itemId: TestData.validCheckoutObject().itemId,
+    };
+    const res = await this.chai
+      .request(this.app)
+      .post(this.apiEndpoint)
+      .set('idToken', idToken)
+      .set('content-type', 'application/json')
+      .send(parameters);
+
+    // THEN: Returns a well formed response
+    super.assertRequestFormat(res, 'Error', 400, 'Error');
+    // THEN: Failed to validate input
+    this.expect(res.body.body.data).to.deep.equal({ message: 'Could not retrieve user ID from provided information' });
+  }
+
+  @test('fails to return checkout object due to no item')
+  @slow(1500)
+  public async returnCheckoutFailsDueToNoItemId() {
```

Comment:
Duplicate?

Reply:
For which one?

Reply:
I saw the same function defined a bit earlier in the file. Did I misread that?

Reply:
The one earlier also tests returning an item but that one succeeds. This one tests when the user tries to return an item without putting an item in which should return with an error. All the tests are separated into successful api requests above and then all the failed api requests that return an error below.